appointment/src/main/java/com/bmwapp/appointment/service/AppointmentService.java

```diff
@@ -63,15 +71,37 @@ public Appointment updateAppointment(Appointment appointment) {
         Provider<Appointment> appointmentProvider = p -> oldAppointment;
         propertyMapper.setProvider(appointmentProvider);
 
+        if(contextProvider.userHasRole("CUSTOMER")){
+            modelMapper.getTypeMap(Appointment.class, Appointment.class)
+                    .addMappings(mapper ->mapper.skip(Appointment::setCustomerId))
+                    .addMappings(mapper ->mapper.skip(Appointment::setFlatId))
+                    .addMappings(mapper ->mapper.skip(Appointment::setAppointmentName));
+        } else {
+            modelMapper.getTypeMap(Appointment.class, Appointment.class)
+                    .addMappings(mapper ->mapper.map(Appointment::getCustomerId, Appointment::setCustomerId))
+                    .addMappings(mapper ->mapper.map(Appointment::getFlatId, Appointment::setFlatId))
+                    .addMappings(mapper ->mapper.map(Appointment::getAppointmentName, Appointment::setAppointmentName));
+        }
+
         Appointment updatedAppointment = modelMapper.map(appointment, Appointment.class);
         return appointmentRepository.save(updatedAppointment);
     }
 
-    public void deleteAppointmentByIdAndCustomerId(Integer appointmentId, String customerId) {
-        Appointment appointment = appointmentRepository
-                .findById(appointmentId)
-                .orElseThrow(() -> new ResourceNotFoundException(String.format("Appointment with id %d not found", appointmentId)));
-        appointmentRepository.deleteByIdAndCustomerId(appointmentId, customerId);
+    public void deleteAppointment(Integer appointmentId) {
+        if (!appointmentRepository.findById(appointmentId).isPresent() || (contextProvider.userHasRole("CUSTOMER")
+                && !appointmentRepository.findOneByIdAndCustomerId(appointmentId, contextProvider.getLoggedUserId()).isPresent())) {
+            throw new ResourceNotFoundException(String.format("Appointment not found"));
+        }
+
+        if(contextProvider.userHasRole("CUSTOMER")){
+            appointmentRepository.deleteByIdAndCustomerId(appointmentId, contextProvider.getLoggedUserId());
+        } else {
```

Comment:
can we make it as a `else if(role === "AGENT")`?

Reply:
Chyba można. Nie mamy więcej ról więc dałem else.